source/common/http/conn_manager_utility.cc

```diff
@@ -165,6 +165,10 @@ ConnectionManagerUtility::MutateRequestHeadersResult ConnectionManagerUtility::m
     request_headers.setReferenceForwardedProto(connection.ssl() ? Headers::get().SchemeValues.Https
                                                                 : Headers::get().SchemeValues.Http);
   }
+  if (config.schemeToSet().has_value()) {
+    request_headers.setScheme(config.schemeToSet().value());
+    request_headers.setForwardedProto(config.schemeToSet().value());
```

Comment:
Technically if we control the values here (enum, etc.) I think we could use reference variants here.

Reply:
Shouldn't `x-forwarded-proto` be the old one, not the new one?

Reply:
I think in the case we force-overwrite we should force-overwrite both, which is what we have in the API and docs.

Reply:
Hmm,  I am not sure.   I thought XFP should be the one from original client,  force-overwrite should not overwrite it.

Reply:
Scheme should also be from the original client. :-)
If we're using this knob it's cause we want to set a scheme which doesn't match underlying encryption.

I'm going for the principle of least surprise here.  Because XFP is the way to communicate http/https for HTTP/1.1, if we have several Envoy hops of mixed HTTP/1 and HTTP/2 we want to force something to look like it's https scheme, you must overwrite both scheme and XFP to have it consistently look like it's https scheme through both sets of hops.  If folks want scheme and XFP to not match they can always change XFP back in request_headers_to_add, but I assume they're usually going to want to have them match.

Reply:
My understanding is: scheme should always matches underlying connection security level.  I could be wrong then.  In that case,  I am fine overwriting both scheme and XFP

Reply:
ah, :scheme actually indicates a fundamental part of the URI
https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2.3
It's a fundamental part of the resource being requested and is not supposed to change from hop to hop based on connection security level.